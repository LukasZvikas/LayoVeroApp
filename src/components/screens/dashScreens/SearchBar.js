import React, { Component } from "react";
import { connect } from "react-redux";
import {
  Text,
  View,
  TextInput,
  TouchableOpacity,
  FlatList,
  Image,
  Keyboard
} from "react-native";
import { dashStyle } from "../../../styles/indexAfterLogin";
import { baseStyle } from "../../../styles/base";
import { Search } from "../../svg";
import {
  getCities,
  getCityFromPartialQuery,
  clearSuggestions
} from "../../../actions/routeActions";
import { Suggestions } from "./suggestions";

class SearchBar extends Component {
  constructor(props) {
    super(props);

    this.state = { showSearchIcon: true, searchBarState: false, text: "" };
  }

  handleInputChange = event => {
    if (event.nativeEvent.text && event.nativeEvent.text.length > 0) {
      this.props.getCityFromPartialQuery(event.nativeEvent.text);
    } else {
      this.props.clearSuggestions();
    }
  };

  render() {
    return (
      <View
        style={
          this.state.searchBarState
            ? [
                dashStyle.focusedSearchBarView,
                baseStyle.justifyFlexStart,
                baseStyle.alignCenter
              ]
            : {}
        }
      >
        <View style={[dashStyle.searchBarWrappper, baseStyle.flexRow, baseStyle.centerItems]}>
          {/*SEARCH BAR INPUT*/}
          <View
            style={[
              dashStyle.searchBarView,
              baseStyle.centerItems,
              this.state.searchBarState
                ? { width: 70 + "%" }
                : { width: 80 + "%" }
            ]}
          >
            {this.state.showSearchIcon ? (
              <View style={{ paddingRight: 5 }}>
                <Search color="#686868" />
              </View>
            ) : null}

            <TextInput
              ref={this.myTextInput}
              style={{ width: 80 + "%" }}
              placeholder="Try London"
              onFocus={() => {
                this.setState({ searchBarState: !this.state.searchBarState });
              }}
              value={this.state.text}
              onChange={event => {
                this.handleInputChange(event);
                this.setState({ text: event.nativeEvent.text });
              }}
            />
          </View>
          {/**/}
          {/*CANCEL BUTTON WHEN SEARCH BAR IS FOCUSED*/}
          {this.state.searchBarState ? (
            <TouchableOpacity
              style={{
                marginLeft: 20,
                alignItems: "center"
              }}
              onPress={() => {
                this.setState({ text: "" });
                Keyboard.dismiss();
                this.setState({ searchBarState: !this.state.searchBarState });
                this.props.getCities();
                this.props.clearSuggestions();
              }}
            >
              <Text>Cancel</Text>
            </TouchableOpacity>
          ) : null}
          {/**/}
          <Suggestions suggestions={this.props.suggestions} />
        </View>
      </View>
    );
  }
}
const mapStateToProps = state => {
  return {
    layovers: state.routes.routesArray,
    suggestions: state.routes.options
  };
};

export default connect(mapStateToProps, {
  getCities,
  getCityFromPartialQuery,
  clearSuggestions
})(SearchBar);
