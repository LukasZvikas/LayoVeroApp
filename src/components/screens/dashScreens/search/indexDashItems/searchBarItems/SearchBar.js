import React, { Component } from "react";
import { connect } from "react-redux";
import { Keyboard, View } from "react-native";
import { searchBarStyle } from "../../../../../../styles/indexDashboard/searchBar";
import { baseStyle } from "../../../../../../styles/base";
import {
  getCities,
  getCityFromPartialQuery,
  clearSuggestions
} from "../../../../../../actions/routeActions";
import InputIcon from "./inputIcon";
import InputText from "./inputText";
import CancelButton from "./cancelButton";
import Suggestions from "../suggestionItems/suggestions";

export class SearchBar extends Component {
  constructor(props) {
    super(props);

    this.state = { showSearchIcon: true, searchBarState: false, text: "" };
  }

  setTextState = event => {
    this.setState({ text: event.nativeEvent.text });
  };
  setSearchState = state => {
    this.setState({ searchBarState: !this.state.searchBarState });
  };
  clearTextState = () => {
    return this.setState({ text: "" });
  };

  changeSearchState = () => {
    this.setState({ searchBarState: !this.state.searchBarState, text: "" });
  };

  getCity = city => {
    return this.props.getCities(city);
  };

  clearSuggestions = () => {
    return this.props.clearSuggestions();
  };

  keyboardDismiss = () => {
    return Keyboard.dismiss();
  };

  handleInputChange = event => {
    if (event.nativeEvent.text && event.nativeEvent.text.length > 0) {
      this.props.getCityFromPartialQuery(event.nativeEvent.text);
    } else {
      this.props.clearSuggestions;
    }
  };

  render() {
    const { clearSuggestions, suggestions, getCities } = this.props;
    return (
      <View
        style={
          this.state.searchBarState
            ? [
                searchBarStyle.focusedSearchBarView,
                baseStyle.justifyFlexStart,
                baseStyle.alignCenter
              ]
            : {}
        }
      >
        <View
          style={[
            searchBarStyle.searchBarWrappper,
            baseStyle.flexRow,
            baseStyle.centerItems
          ]}
        >
          <View
            style={[
              searchBarStyle.searchBarView,
              baseStyle.centerItems,
              this.state.searchBarState
                ? { width: 70 + "%" }
                : { width: 80 + "%" }
            ]}
          >
            <InputIcon showSearchIcon={this.state.showSearchIcon} />
            <InputText
              onChange={event => {
                this.handleInputChange(event);
                this.setTextState(event);
              }}
              textState={this.state.text}
              onFocus={() => {
                this.setSearchState();
              }}
            />
          </View>
          <CancelButton
            searchBarState={this.state.searchBarState}
            onPress={() => {
              this.clearTextState();
              this.keyboardDismiss();
              this.clearSuggestions();
              this.getCity("London");
              this.setSearchState();
            }}
          />
          <Suggestions
            suggestions={suggestions}
            setSearchBarState={() => this.changeSearchState()}
            clearSuggestions={() => this.clearSuggestions()}
            dismissKeyboard={() => this.keyboardDismiss()}
          />
        </View>
      </View>
    );
  }
}
const mapStateToProps = state => {
  return {
    layovers: state.routes.routesArray,
    suggestions: state.routes.options
  };
};

export default connect(mapStateToProps, {
  getCities,
  getCityFromPartialQuery,
  clearSuggestions
})(SearchBar);
