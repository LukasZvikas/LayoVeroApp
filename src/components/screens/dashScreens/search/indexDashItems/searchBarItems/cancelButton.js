import React from "react";
import { Text, TouchableOpacity, StyleSheet } from "react-native";

export const styles = StyleSheet.create({
  wrapper: {
    marginLeft: 20,
    alignItems: "center"
  }
});
const CancelButton = ({ searchBarState, onPress }) => {
  return searchBarState ? (
    <TouchableOpacity style={styles.wrapper} onPress={onPress}>
      <Text>Cancel</Text>
    </TouchableOpacity>
  ) : null;
};

export default CancelButton;
