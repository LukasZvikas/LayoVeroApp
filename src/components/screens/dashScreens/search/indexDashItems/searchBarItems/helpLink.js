import React, { Component } from "react";

import { Text, TouchableOpacity } from "react-native";
import { searchBarStyle } from "../../../../../../styles/indexDashboard/searchBar";

const HelpLink = ({ action }) => {
  return (
    <TouchableOpacity onPress={action}>
      <Text style={searchBarStyle.helpText}>Need Help Looking?</Text>
    </TouchableOpacity>
  );
};

export default HelpLink;
