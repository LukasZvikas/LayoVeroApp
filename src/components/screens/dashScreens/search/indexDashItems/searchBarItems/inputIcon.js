import React from "react";
import { View, StyleSheet } from "react-native";
import { Search } from "../../../../../svg";

export const styles = StyleSheet.create({
  padding: {
    paddingRight: 5
  }
});

const InputIcon = ({ showSearchIcon }) => {
  return showSearchIcon ? (
    <View style={styles.padding}>
      <Search color="#686868" />
    </View>
  ) : null;
};

export default InputIcon;
