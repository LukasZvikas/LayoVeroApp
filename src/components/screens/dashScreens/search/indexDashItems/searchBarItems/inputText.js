import React from "react";
import { TextInput, StyleSheet } from "react-native";

export const styles = StyleSheet.create({
  inputWidth: { width: 80 + "%" }
});

const InputText = ({ onFocus, textState, onChange }) => {
  return (
    <TextInput
      style={styles.inputWidth}
      placeholder="Try London"
      onFocus={onFocus}
      value={textState}
      onChange={onChange}
    />
  );
};

export default InputText;
