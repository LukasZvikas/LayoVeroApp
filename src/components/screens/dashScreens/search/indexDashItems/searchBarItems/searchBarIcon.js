import React from "react";
import { View, StyleSheet } from "react-native";
import { searchBarStyle } from "../../../../../../styles/indexDashboard/searchBar";
import { baseStyle } from "../../../../../../styles/base";
import { Search } from "../../../../../svg";

export const styles = StyleSheet.create({
  padding: {
    paddingRight: 5
  }
});

const SearchBarIcon = ({ showSearchIcon }) => {
  return showSearchIcon ? (
    <View style={styles.padding}>
      <Search color="#686868" />
    </View>
  ) : null;
};

export default SearchBarIcon;
