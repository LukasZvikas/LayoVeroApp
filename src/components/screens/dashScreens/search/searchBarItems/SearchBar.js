import React, { Component } from "react";
import { connect } from "react-redux";
import {
  Text,
  View,
  TextInput,
  TouchableOpacity,
  FlatList,
  Image,
  Keyboard
} from "react-native";
import { searchBarStyle } from "../../../../../styles/indexDashboard/searchBar";
import { baseStyle } from "../../../../../styles/base";
import { Search } from "../../../../svg";
import {
  getCities,
  getCityFromPartialQuery,
  clearSuggestions
} from "../../../../../actions/routeActions";
import Suggestions from "./suggestions";

class SearchBar extends Component {
  constructor(props) {
    super(props);

    this.state = { showSearchIcon: true, searchBarState: false, text: "" };
  }

  changeSearchState = () => {
    this.setState({ searchBarState: !this.state.searchBarState, text: "" });
  };

  handleInputChange = event => {
    if (event.nativeEvent.text && event.nativeEvent.text.length > 0) {
      this.props.getCityFromPartialQuery(event.nativeEvent.text);
    } else {
      this.props.clearSuggestions();
    }
  };

  render() {
    const { clearSuggestions, suggestions, getCities } = this.props;
    return (
      <View
        style={
          this.state.searchBarState
            ? [
                searchBarStyle.focusedSearchBarView,
                baseStyle.justifyFlexStart,
                baseStyle.alignCenter
              ]
            : {}
        }
      >
        <View
          style={[
            searchBarStyle.searchBarWrappper,
            baseStyle.flexRow,
            baseStyle.centerItems
          ]}
        >
          {/*SEARCH BAR INPUT*/}
          <View
            style={[
              searchBarStyle.searchBarView,
              baseStyle.centerItems,
              this.state.searchBarState
                ? { width: 70 + "%" }
                : { width: 80 + "%" }
            ]}
          >
            {this.state.showSearchIcon ? (
              <View style={{ paddingRight: 5 }}>
                <Search color="#686868" />
              </View>
            ) : null}

            <TextInput
              ref={this.myTextInput}
              style={{ width: 80 + "%" }}
              placeholder="Try London"
              onFocus={() => {
                this.setState({ searchBarState: !this.state.searchBarState });
              }}
              value={this.state.text}
              onChange={event => {
                this.handleInputChange(event);
                this.setState({ text: event.nativeEvent.text });
              }}
            />
          </View>
          {/**/}
          {/*CANCEL BUTTON WHEN SEARCH BAR IS FOCUSED*/}
          {this.state.searchBarState ? (
            <TouchableOpacity
              style={{
                marginLeft: 20,
                alignItems: "center"
              }}
              onPress={() => {
                this.setState({ text: "" });
                Keyboard.dismiss();
                clearSuggestions();
                getCities("London");
                this.setState({ searchBarState: !this.state.searchBarState });
              }}
            >
              <Text>Cancel</Text>
            </TouchableOpacity>
          ) : null}
          {/**/}
          <Suggestions
            suggestions={suggestions}
            setSearchBarState={() => this.changeSearchState()}
            clearSuggestions={() => clearSuggestions()}
            dismissKeyboard={() => Keyboard.dismiss()}
          />
        </View>
      </View>
    );
  }
}
const mapStateToProps = state => {
  return {
    layovers: state.routes.routesArray,
    suggestions: state.routes.options
  };
};

export default connect(mapStateToProps, {
  getCities,
  getCityFromPartialQuery,
  clearSuggestions
})(SearchBar);
