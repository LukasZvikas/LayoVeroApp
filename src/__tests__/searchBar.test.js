import React from "react";
import { shallow, mount } from "enzyme";
import SearchBar from "../components/screens/dashScreens/search/indexDashItems/searchBarItems/SearchBar";
import { SearchBarIcon } from "../components/screens/dashScreens/search/indexDashItems/searchBarItems/searchBarIcon";
import Root from "../../root";
import renderer from "react-test-renderer";

it("shows SearchBarIcon", () => {
  const wrapper = renderer
    .create(
      <Root>
        <SearchBarIcon />
      </Root>
    )
    .toJSON();

  expect(wrapper).toMatchSnapshot();
});
