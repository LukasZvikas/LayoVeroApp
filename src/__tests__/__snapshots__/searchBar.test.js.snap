// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shows SearchBarIcon 1`] = `null`;
